### DISCUSSION TITLE
comments in auto-generated kprobe_bpf.go file causing errors

### DISCUSSION THREAD

**Discussion by camrynl** [cilium/ebpf#695] [Q&A] [answered]
When trying to test my project, an error is raised from the auto-generated kprobe_bpf.go file. 

![image](https://user-images.githubusercontent.com/31013536/171699580-cc072238-bf83-4715-b4fd-aac238be3d28.png)

It's coming from this line, and seems to go away when I fix the comment by adding a space to change the comment to "// go..." This auto-generated file says "DO NOT EDIT" at the top. Will modifying this comment in the file cause any issues?

**Comment by ti-mo** [comment, marked as answer]
@camrynl First, please read up on what `go:embed` is: https://blog.jetbrains.com/go/2021/06/09/how-to-use-go-embed-in-go-1-16.

I think the error is rather clear: `kprobe_bpf.o` does not exist, which means you've likely removed it after running bpf2go or forgotten to check it into version control. For most use cases, we recommend checking in the BPF object since it means your package becomes `go get`table and will not require LLVM or bpf2go to be installed on your CI infrastructure.

> seems to go away when I fix the comment by adding a space to change the comment to "// go..."

Adding the space indeed makes the compiler treat the annotation as a normal comment and will thus no longer embed the file during compilation.

> Will modifying this comment in the file cause any issues?

No, but it will be overwritten by bpf2go next time you rebuild your eBPF program.

> Will modifying this comment in the file cause any issues?

Your program won't contain any BPF bytecode and so will fail to be loaded into the kernel. I think that qualifies as problematic. :slightly_smiling_face:
